### PR TITLE
Adding conv2d operation

### PR DESCRIPTION
**Pull request by taless474**
This PR adds conv2d which imitates the functionality of Keras [conv2d](https://keras.io/backend/#conv2d).
It also modifies conv1d for the cases that the image is shorter than the filter having `same` padding.
Toward solving #857

### CODE REVIEW

**Comment by hkaiser** [comment]
This was superseded by #1042
